src/main/java/com/jcabi/xml/ClasspathInput.java

```diff
@@ -0,0 +1,160 @@
+ * NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jcabi.xml;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Reader;
+import org.apache.commons.io.IOUtils;
+import org.w3c.dom.ls.LSInput;
+
+/**
+ * {@link org.w3c.dom.ls.LSInput} implementation
+ * used by {@link com.jcabi.xml.ClasspathResolver}.
+ *
+ * @author Adam Siemion (adam.siemion.null@lemonsoftware.pl)
+ * @version $Id$
+ */
+class ClasspathInput implements LSInput {
+
+    /**
+     * Input stream.
+     */
+    private final transient BufferedInputStream stream;
+    /**
+     * Public Id.
+     */
+    private transient String publicid;
+    /**
+     * System Id.
+     */
+    private transient String systemid;
+
+    /**
+     * Public constructor.
+     * @param pubid Public id
+     * @param sysid System id
+     * @param input Input stream
+     */
+    public ClasspathInput(
+        final String pubid, final String sysid, final InputStream input
+    ) {
+        this.publicid = pubid;
+        this.systemid = sysid;
+        this.stream = new BufferedInputStream(input);
+    }
+
+    @Override
+    public String getPublicId() {
+        return this.publicid;
+    }
+
+    @Override
+    public void setPublicId(final String pubid) {
+        this.publicid = pubid;
+    }
+
+    @Override
+    public String getSystemId() {
+        return this.systemid;
+    }
+
+    @Override
+    public void setSystemId(final String sysid) {
+        this.systemid = sysid;
+    }
+
+    @Override
+    public String getBaseURI() {
+        return null;
+    }
+
+    @Override
+    public InputStream getByteStream() {
+        return null;
+    }
+
+    @Override
+    @SuppressWarnings("PMD.BooleanGetMethodName")
+    public boolean getCertifiedText() {
+        return false;
+    }
+
+    @Override
+    public Reader getCharacterStream() {
+        return null;
+    }
+
+    @Override
+    public String getEncoding() {
+        return null;
+    }
+
+    @Override
+    @SuppressWarnings("PMD.AvoidThrowingRawExceptionTypes")
+    public String getStringData() {
+        try {
+            return IOUtils.toString(this.stream);
+        } catch (final IOException exception) {
+            throw new RuntimeException("Unable to read input", exception);
```

Comment:
why throw RuntimeException?
Throw IllegalArgumentException and remove SuppressWarnings